index.js

```diff
@@ -2,14 +2,14 @@ const { dialog, globalShortcut } = require('electron');
 const getShortcutFromConfig = require('./modules/getShortcutFromConfig');
 
 function registerShortcut (configKey, action) {
-  return (app) => {
+  return (app, ...args) => {
     const { plugins, config } = app;
     function register (accelerator) {
       if (!accelerator) return;
       globalShortcut.unregister(accelerator);
       const registered = globalShortcut.register(
         accelerator,
-        () => action(app)
+        () => action.apply(this, [app, ...args])
```

Comment:
Correct, `this` is somewhat irrelevant, but I think `apply` is necessary here.

I tried your proposed approached, but the "remaining" arguments then come through as an array. So, the called function (in this case `toggleWindowVisibility`) would have to destructure the "remaining" arguments from the array. My goal was for the called function to end up with the originally passed arguments in the same order.

To explain in examples...

### Using `apply`
```javascript
// Registration
registerShortcut('summon', toggleWindowVisibility)(app, windowSet);

// Passing arguments through
return (app, ...args) => {
  // [...]
  () => action.apply(this, [app, ...args])
  // [...]
}

// Utilizing passed arguments
toggleWindowVisibility (app, windowSet) {
  // Here `windowSet` is the originally passed `windowSet`
}
```

### Simple pass through
```javascript
// Registration
registerShortcut('summon', toggleWindowVisibility)(app, windowSet);

// Passing arguments through
return (app, ...args) => {
  // [...]
  () => action(app, args)
  // [...]
}

// Utilizing passed arguments
toggleWindowVisibility (app, windowSet) {
  // Here `windowSet` is an array of the "remaining" original arguments
  // (i.e. `[windowSet, arg3, arg4, etc]`)
}
```

I'm only passing one argument (`windowSet`), but in the future someone may want to pass several. I figured the "remaining" arguments coming through as an array would likely be super confusing in that context. It would require the user to destructure the "remaining" arguments, even though they originally passed them in individually.